frontend/vite.config.js

```diff
@@ -1,7 +1,21 @@
 import { defineConfig } from 'vite'
 import vue from '@vitejs/plugin-vue'
+import basicSsl from '@vitejs/plugin-basic-ssl'
 
 // https://vitejs.dev/config/
 export default defineConfig({
-  plugins: [vue()],
+  plugins: [
+    vue(),
+    basicSsl({
+      name: 'yamo'
+    })
+  ],
+  server: {
+    proxy: {
```

Comment:
will redirect https://<your-local-ip>:5173/api  -> http://localhost:3000/ where the local API is running.